src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+            }
+            else if ( read.getContig() == read.getMateContig() )
+            {
+                if ( read.isReverseStrand() == read.mateIsReverseStrand() || !read.isProperlyPaired() )
+                {
+                    int locus = read.getFragmentLength() < 0 ? read.getStart() - FUNKY_PAIR_LOCUS_WIDTH : read.getEnd();
+                    return new EventLocus(locus,FUNKY_PAIR_LOCUS_WIDTH);
+                }
+            }
+
+            return null;
+        }
+    }
+
+    // A class that acts as a filter for a stream of reads.  It passes only those that are a part of a putative
+    // breakpoint cluster.
+    private static final class BreakpointClusterer implements Iterator<GATKRead>, Iterable<GATKRead>
+    {
+        private SplitReadDetector splitReadDetector;
+        private DiscordantPairDetector discordantPairDetector;
+        private final Iterator<GATKRead> inputIterator;
+        private String currentContig = null;
+        private long eventIdCounter = 0;
+        private final SortedMap<EventLocus,GATKRead> locMap = new TreeMap<>();
+        private Iterator<GATKRead> outputIterator = Collections.emptyIterator();
+
+        private static final int MIN_MATCH_LEN = 45; // minimum length of matched portion of an interesting alignment
+        private static final int MIN_MAPQ = 20; // minimum mapping quality of an interesting alignment
+        private static final int MAX_LOCUS_DIST = 300; // stale evidence distance, should be about the fragment length
+        private static final int MIN_EVIDENCE = 5; // minimum evidence count in a cluster
+
+        public BreakpointClusterer( final SplitReadDetector someSplitReadDetector,
+                                    final DiscordantPairDetector someDiscordantPairDetector,
+                                    final Iterator<GATKRead> someInputIterator )
+        {
+            splitReadDetector = someSplitReadDetector;
+            discordantPairDetector = someDiscordantPairDetector;
+            inputIterator = someInputIterator;
+        }
+
+        @Override
+        public boolean hasNext()
+        {
+            while ( !outputIterator.hasNext() )
+            {
+                if ( !inputIterator.hasNext() )
+                    return false;
+                outputIterator = processRead(inputIterator.next());
+            }
+            return true;
+        }
+
+        @Override
+        public GATKRead next()
+        {
+            if ( !hasNext() )
+                throw new NoSuchElementException("Iterator<GATKRead> is exhausted.");
+            return outputIterator.next();
+        }
+
+        @Override
+        public Iterator<GATKRead> iterator() { return this; }
+
+        private Iterator<GATKRead> processRead( final GATKRead read )
+        {
+            if ( !read.isUnmapped() &&
+                    read.getMappingQuality() >= MIN_MAPQ &&
+                    read.getStart() != ReadConstants.UNSET_POSITION )
```

Comment:
Did you see a read that isn't unmapped but but has an unset start position?   I was under the impression that this is not possible.